lab-maddy/src/action/expense-actions.js

```diff
@@ -0,0 +1,24 @@
+import uuid from 'uuid/v4';
+
+export const expenseCreate = expense => { //an action creator
+  expense.id = uuid();
+  expense.timestamp = new Date();
+  return {
+    type: 'EXPENSE_CREATE',
+    payload: expense,
+  };
+};
+
+export const expenseUpdate = expense => {
+  return {
+    type: 'EXPENSE_UPDATE',
+    payload: expense,
+  };
+};
+
+export const cardDelete = expense => {
```

Comment:
on line 12 you've changed cardUpdate to expenseUpdate, but cardDelete is still card. If you change one, you need to change all, because it's confusing